jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -139,7 +148,9 @@ public void preConfigure(WebAppContext context) throws Exception
                 if (quickStartWebXml.exists())
                     quickStart(context);
                 else
-                    throw new IllegalStateException("No " + quickStartWebXml);
+                {
```

Comment:
I don't think it's appropriate to skip the deployment if `QUICKSTART` mode is selected.  In a mixed-context environment, you can easily set the context attribute to `AUTO` for those contexts for which quickstart isn't appropriate. I think it's a real error if you've selected `QUICKSTART` mode, but yet no `quickstart-web.xml` file exists.

Reply:
This would require every context to be supplemented by an XML deployable.
Why isn't the default set to AUTO?
Let me try that as default and see what happens in the test case.

Reply:
AUTO is the default if unset.
But the Configuration field approach was wrong, as that is keeping track of the Mode in two places.
If the mode is specified in start.d/quickstart.ini then it's automatically set on the context.attributes.
I reworked this to not have the Configuration field too, as the existing context.attribute approach is correct.